vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java

```diff
@@ -754,18 +754,24 @@ private void onTargetAttach(UI ui) {
     }
 
     /**
-     * Finds the closest parent component that is annotated with
-     * {@link ModalRoot}, if any.
+     * Finds the closest candidate in the component hierarchy to use as a parent
+     * component when auto-adding the popover. This can be:
+     * <ul>
+     * <li>Another popover component</li>
+     * <li>A component annotated with {@link ModalRoot}</li>
+     * </ul>
      *
-     * @return an optional with the closest modal parent component, or an empty
-     *         optional if none found
+     * @return an optional with the candidate to use as parent component, or an
+     *         empty optional if none found
      */
-    private Optional<Component> getModalParentComponent() {
+    private Optional<Component> getParentComponentCandidate() {
         var parent = target.getParent();
         while (parent.isPresent()) {
             var parentComponent = parent.get();
-            if (parentComponent.getClass()
-                    .isAnnotationPresent(ModalRoot.class)) {
+            var isPopover = parentComponent instanceof Popover;
+            var isModal = parentComponent.getClass()
+                    .isAnnotationPresent(ModalRoot.class);
+            if (isPopover || isModal) {
```

Comment:
Chose this approach over adding `ModalRoot` to `Popover`, otherwise it might have required redefining the interface to mean something else. Can maybe reconsider this if we have more use-cases around auto adding something to overlays in general rather than modals specifically.